src/pybiocfilecache/BiocFileCache.py

```diff
@@ -131,7 +159,17 @@ def _get(self, session: Session, rname: str) -> Optional[Resource]:
         Returns:
             res (Resource, optional): The `Resource` for the `rname` if any.
         """
-        return session.query(Resource).filter(Resource.rname == rname).first()
+        resource: Optional[Resource] = (
+            session.query(Resource).filter(Resource.rname == rname).first()
+        )
+
+        if resource is not None:
+            # `Resource` may exist but `rpath` could still be being
+            # moved/copied into by `add`, wait until `rpath` exists
+            while not Path(str(resource.rpath)).exists():
+                sleep(0.1)
```

Comment:
this is an infinite loop until `rpath` exists, there should be a timeout

Reply:
Yeah, the file could be large and take a while in `copy` mode, so 30 seconds?